### PR TITLE
adding common-jwt module

### PR DESCRIPTION
**Pull request by AlexKazmirchuk**
added TokenService interface and his implementation TokenServiceImpl and configuration class for spring
I use io.jsonwebtoken library for creating tokens

### CODE REVIEW

File: softgroup/basis/common-jwt/src/main/java/com/softgroup/common/jwt/api/TokenService.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    String getProfileID(String token);
+
+    String getDeviceID(String token);
+
+    Long getCreationTime(String token);
+
+    Long getExpirationTime(String token);
+
+    String getTokenType(String token);
```
Comment (Dominys) [inline softgroup/basis/common-jwt/src/main/java/com/softgroup/common/jwt/api/TokenService.java]: @AlexKazmirchuk Lets make type Enum

**Comment by Dominys** [comment]
@AlexKazmirchuk Where is tests ?

File: softgroup/basis/common-jwt/src/main/java/com/softgroup/common/jwt/api/TokenService.java (hunk 1 above)
Comment (AlexKazmirchuk) [reply softgroup/basis/common-jwt/src/main/java/com/softgroup/common/jwt/api/TokenService.java]: @Dominys done! i think

**Comment by AlexKazmirchuk** [comment]
@Dominys Added tests

File: softgroup/basis/common-jwt/src/main/java/com/softgroup/common/jwt/api/TokenService.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import io.jsonwebtoken.SignatureException;
+
+/**
+ * @author AlexKazmirchuk
+ * @since 16.03.17.
+ */
+
+public interface TokenService {
+
+    void validateSessionToken(String sessionToken) throws ExpiredJwtException, MalformedJwtException, SignatureException;
```
Comment (Dominys) [inline softgroup/basis/common-jwt/src/main/java/com/softgroup/common/jwt/api/TokenService.java]: @AlexKazmirchuk Throw some one custom exception.

Comment (AlexKazmirchuk) [inline softgroup/basis/common-jwt/src/main/java/com/softgroup/common/jwt/api/TokenService.java]: @Dominys  for example instead " ExpiredJwtException, MalformedJwtException, SignatureException"

I must throw my custom "InvalidTokenException" ?  

or method signature must be something like : " ExpiredJwtException, MalformedJwtException, SignatureException, InvalidTokenException" ?

Comment (AlexKazmirchuk) [reply softgroup/basis/common-jwt/src/main/java/com/softgroup/common/jwt/api/TokenService.java]: @Dominys done! added custom exception

**Review by Dominys — APPROVED** [review]

**Pull request merged**